R/zchunk_L2522.ag_MACC_IRR_MGMT.R

```diff
@@ -34,117 +33,164 @@ module_emissions_L2522.ag_MACC_IRR_MGMT_DISABLED <- function(command, ...) {
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+
+    # Add irrigation management level to the marginal abatement cost (MAC) curves for agriculture.
+    L2521.AgMAC %>%
+      repeat_add_columns(tibble(lvl = c("lo", "hi"))) %>%
+      unite(AgProductionTechnology, AgProductionTechnology, lvl, sep = "_") ->
+      L2522.AgMAC
+
+
+    # Add technology change data to the agricultural and animal curve technology MAC curves by
+    # SSP scenario and mac.control (supplysector) then create separate data frames for each SPP MAC curve.
+
+    # First save a list of all the SSP-sepcific scenarios in the from the MAC curve technology change parameters
+    # assumption file.
+    A_MACC_TechChange %>%
+      select(scenario) %>%
+      distinct->
+      scenario_list
+
+    # Add SSP scenarios to the the MAC agriculture data frame.
+    L2522.AgMAC %>%
+      repeat_add_columns(scenario_list) ->
+      L2522.AgMAC_all_SSP
+
+    # Add SSP scenarios to the the MAC animal data frame.
+    L252.MAC_an %>%
+      repeat_add_columns(scenario_list) ->
+      L2522.MAC_an_all_SSP
+
+
+    # Add the technology change to the agricultural and animal MAC curves by matching SSP
+    # scenario and mac.control (spplysector).
     #
+    # Agricultural MAC curve
+    L2522.AgMAC_all_SSP %>%
+      left_join_error_no_match(A_MACC_TechChange %>% select(scenario, MAC, tech_change),
+                               by = c("scenario", c("mac.control" = "MAC"))) %>%
+      select(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology, year, Non.CO2, mac.control,
+             tax, mac.reduction, tech.change = tech_change, scenario) ->
+      L2522.AgMAC_all_SSP_tech.change
+
+
+    # Animal MAC curve
+    L2522.MAC_an_all_SSP %>%
+      left_join_error_no_match(A_MACC_TechChange %>% select(scenario, MAC, tech_change),
+                               by = c("scenario", c("mac.control" = "MAC"))) %>%
+      select(region, supplysector, subsector, stub.technology, year, Non.CO2, mac.control,
+             tax, mac.reduction, tech.change = tech_change, scenario) ->
+      L2522.MAC_an_all_SSP_tech.change
+
+
+    # Separate the MAC curves by SPP scenario.
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Agricultural MAC curve for SSP1
+    L2522.AgMAC_all_SSP_tech.change %>%
```

Comment:
Perhaps making a little `filter_ssp` function that filters and drops `scenario`, and then calling it six times.

Reply:
I.e.

```R
filter_spp <- function(x, ssp) {
  x %>%
       filter(scenario == ssp) %>%
       select(-scenario) 
}
```